net/golioth/golioth.c

```diff
@@ -532,8 +532,16 @@ int golioth_process_rx(struct golioth_client *client)
 	int ret;
 	int err;
 
+	/* nRF9160 does not support ZSOCK_MSG_TRUNC */
+#if CONFIG_NET_SOCKETS_OFFLOAD && CONFIG_NET_SOCKETS && CONFIG_NRF_MODEM_LIB
+	int flags = ZSOCK_MSG_DONTWAIT;
+#else 
+	int flags = ZSOCK_MSG_DONTWAIT | ZSOCK_MSG_TRUNC;
+#endif
+
```

Comment:
I'd prefer to introduce new Kconfig option like:
```
config GOLIOTH_RECV_USE_MSG_TRUNC
        bool "Use MSG_TRUNC in recv()"
        default y if NET_NATIVE && !NET_SOCKES_OFFLOAD
        help
          Use MSG_TRUNC flag in recv() API calls. This allows to receive truncated
          packets and renegotiate smaller blocks in CoAP packets in case configured
          receive buffer is too small.
```

Then we can use `IS_ENABLED(CONFIG_GOLIOTH_RECV_USE_MSG_TRUNC)` to construct `flags` without too much ifdef'fery in C code.